src/http/server/handlers/static_file_handler.cr

```diff
@@ -99,9 +101,55 @@ class HTTP::StaticFileHandler
         end
       end
 
-      context.response.content_length = file_info.size
       File.open(file_path) do |file|
-        IO.copy(file, context.response)
+        if (range = context.request.headers["Range"]?) && (match = range.match(/bytes=([0-9\-,]+)/))
+          ranges = match[1].split(',').map do |range|
+            start, finish = range.strip.split('-', 2)
+            start = start.to_i64 { 0 }
+            finish = finish.to_i64 { file_info.size }
+
+            {start, finish}
```

Comment:
Why not use a `Range` here? Seems like a perfect fit.
Size calculation is already included for free.